dttools/src/catalog_query.c

```diff
@@ -227,12 +229,28 @@ int catalog_query_send_update(const char *hosts, const char *text)
 	if (!d) {
 		fatal("could not create datagram port!");
 	}
+	
+	data_length = strlen(text)+1;
+
+	/* Estimates the bounds for the compressed data. */
+	compress_data_length = compressBound(data_length);
+	char* compress_data= malloc(compress_data_length);
+
+	/* Prefix the data with 0x1A (Control-Z) to indicate a compressed packet. */
+	compress_data[0] = 0x1A;
+
+	int success = compress((Bytef*)compress_data+1, &compress_data_length, (const Bytef*)text, data_length);
```

Comment:
compress_data_length-1?

Reply:
compress_data_length is an estimated value from compressBound and is then set by the compress function. I don't modify it here as I will modify it when we send it.